docs/schema.md

```diff
@@ -13,317 +19,465 @@ This schema document describes the schemas for each part in this stack:
-        /* Time window size in milliseconds */
-        "timeExtent": 10368057344, 
-
-        /* Screen center and zoom, google maps semantics */
-        "lon": -169.69070434570312, 
-        "lat": -7.453709339338128, 
-        "zoom": 4, 
+## Animaton configuration
 
-        /* Animation state */
-        "paused": true, 
-        "loop": true,
-      },
-      "map": {
-        /* Google maps styling and other options */
-        "options": {
-          "mapTypeId": "roadmap", 
-          "styles": [
-            {"featureType": "road", "stylers": [{"visibility": "off"}]}
-          ]
+Animations represents layers of data overlaid on the basemap. Each animation
+visualizes some source data. There is no hard-coded schema for source data.
+Instead, the animation configuration dynamically maps source data to the values
+needed by the visualization. The user may also dynamically change this mapping
+using sliders or buttons in the editing UI or a custom UI.
+
+Each animation definition contains a title, a type and a set of arguments. The
+arguments contain at least a source definition, with a source type and loader
+arguments.
+
+```
+{
+  "type": "MyAnimationClass"
+
+  "args": {
+
+    "title": "My animation"
+
+    /* Visibility checkbox status */
+    "visible": true,
+
+    /* Indicates that this layer supports reporting, allowing other layers to
+       run reports related to this one. For example, you may have a vessel activity
+       animation which displays the locations of different vessels, and a EEZ layer
+       which displays the EEZ boundaries. In that case, you may have a report that
+       shows the vessels inside each EEZ. To support this, you would configure the
+       vessel activity layer as reportable, and the EEZ animation would report fields
+       indicating how the report is displayed and generated. */
+    "reportable": true,
+
+    /* Settings for report generation. When an animation has these settings,
+       it allows the user to generate a report over another animation, marked with the
+       `reportable` property. */
+    "report": {
+      /* This field configures the way selected polygon fields are processed
+         for the url and prompt templates */
+      "polygonFields": {
+        /* Here we are making the `boundary` field available in the templates.
+           We are making it a splitable field, which means that the boundary field in the
+           selected polygon actually contains multiple values, such as "Egypt - Greece",
+           and the user should select one of them, either "Egypt" or "Greece" */
+        "boundary": {
+          "split": "-",
+          "label": "EEZ"
         },
 
-        /* Overlay animation confirguration. See below. */
-        "animations": []
-      }
-    }
+        /* Here we are publishing another field from the polygon, in this case one that
+           is directly available in the templates */
+        "country": {}
```

Comment:
Implementation detail, it was easier this way and I needed it working quickly. I'll change it to copy all the fields from the data once I merge the stuff that has server changes.